graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/shared/GenerateKotlinCode.kt

```diff
@@ -37,7 +37,7 @@ fun generateKotlinCode(
             is BooleanValue -> CodeBlock.of("%L", value.isValue)
             is IntValue -> CodeBlock.of("%L", value.value)
             is StringValue -> CodeBlock.of("%S", value.value)
-            is FloatValue -> CodeBlock.of("%L", value.value)
+            is FloatValue -> CodeBlock.of("%L", value.value.toString())
```

Comment:
@paulbakker , open to suggestions on more canonical (per the codebase style) way of preserving decimal points. E.g. without the change: default value `0.0` becomes `0` in the generated code. With the change it stays `0.0`.